src/core/libmaven/csvreports.cpp

```diff
@@ -26,6 +47,88 @@ CSVReports::~CSVReports() {
     closeFiles();
 }
 
+bool CSVReports::exportGroup(){
+
+    if(_fileName.length() < 4){
+        errorReport="invalid file name";
+        return 0;
+    }
+    string fileFormat = _fileName.substr(_fileName.length()-4 , 4 );
+    if( fileFormat == ".csv" )
+        SEP = ",";
+    if( fileFormat == ".tab" )
+        SEP = "\t";
+    outFileStream.open(_fileName);
+    if(!outFileStream.is_open()){
+        errorReport="file can't open to write";
+        return 0;
+    }
+    else{
+        addColumnNames();
+        for(int i=0 ; i<groups.size(); ++i){
+            PeakGroup* group=groups[i];
+
+            if(_exportType == PeakExport)
+                writePeakInfo(group);
+
+            if(_exportType == GroupExport){
+                if( group->children.size() == 0 )
+                    writeGroupInfo(group);
+
+                for( int i = 0 ; i < group->children.size() ; ++i ){
+                    group->children[i].metaGroupId = group->metaGroupId;
+                    writeGroupInfo(&group->children[i]);
+                }      
+            }
+        }
+
+        outFileStream.close();
+    }
+    
+}
+
+void CSVReports::addColumnNames(){
```

Comment:
@shubhra-agrawal It's good practice to not add any redundancy in code. It's just a **if** statement that is controlling otherwise you have to write a whole new method that will add extra load to compiler and linker.

Reply:
@shubhra-agrawal One more point- all columns are getting added at one place so it is more easy to understand the code base. It we add columns at different places, then it will definitely make one to grasp logics behind the columns- What will be the columns name at what conditions.